lib/content/write.js

```diff
@@ -15,6 +15,9 @@ const fsm = require('fs-minipass')
 
 module.exports = write
 
+// Cache of move operations in process so we don't duplicate
+const moveOperations = {}
```

Comment:
my one and only piece of feedback for this is questioning if this is better as an object or a Map. it should get a lot of changes on the object, so i _think_ we would benefit from the faster key lookups and deletions in a Map

Reply:
Yep a map totally makes sense here. They're slightly slower to instantiate but we do that as a singleton so it's immeasurably small.  The repeated lookups will more than pay for themselves.